#!/usr/bin/env python

PACKAGE = "object_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enable_passthrough_filter_x",   bool_t,   0, "enable_passthrough_filter_x",  True)
gen.add("enable_passthrough_filter_y",   bool_t,   0, "enable_passthrough_filter_y",  True)
gen.add("enable_passthrough_filter_z",   bool_t,   0, "enable_passthrough_filter_z",  True)
gen.add("enable_voxel_filter_downsampling",   bool_t,   0, "enable_voxel_filter_downsampling",  True)
gen.add("enable_beforeSeg_statistical_outlier_removal",   bool_t,   0, "enable_beforeSeg_statistical_outlier_removal",  False)
gen.add("enable_ransac_plane_segmentation",   bool_t,   0, "enable_ransac_plane_segmentation",  True)
gen.add("enable_shrinking",   bool_t,   0, "enable_shrinking",  True)
gen.add("enable_afterSeg_statistical_outlier_removal",   bool_t,   0, "enable_afterSeg_statistical_outlier_removal",  True)
gen.add("enable_euclidean_clustering",   bool_t,   0, "enable_euclidean_clustering",  True)
gen.add("enable_3D_bounding_boxes",   bool_t,   0, "enable_3D_bounding_boxes",  True)

gen.add("passthrough_filter_x_low",    double_t,    0, "passthrough_filter_x_low", 0,  -5,  5)
gen.add("passthrough_filter_x_high",    double_t,    0, "passthrough_filter_x_high", 1.6, -5,  5)

gen.add("passthrough_filter_y_low",    double_t,    0, "passthrough_filter_y_low", -1.0, -5,  5)
gen.add("passthrough_filter_y_high",    double_t,    0, "passthrough_filter_y_high", 1.0, -5,  5)

gen.add("passthrough_filter_z_low",    double_t,    0, "passthrough_filter_z_low", -1, -5,  5)
gen.add("passthrough_filter_z_high",    double_t,    0, "passthrough_filter_z_high", 0.25, -5,  5)

gen.add("voxel_filter_downsampling_leaf",    double_t,    0, "voxel_filter_downsampling_leaf",  0.004, 0,  1)

gen.add("beforeSeg_statistical_outlier_removal_K",    double_t,    0, "beforeSeg_statistical_outlier_removal_K", 20,  1, 500)
gen.add("beforeSeg_statistical_outlier_removal_deviation",    double_t,    0, "beforeSeg_statistical_outlier_removal_deviation", 0.01,  0,  1)

gen.add("afterSeg_statistical_outlier_removal_K",    double_t,    0, "afterSeg_statistical_outlier_removal_K", 20,  1, 500)
gen.add("afterSeg_statistical_outlier_removal_deviation",    double_t,    0, "afterSeg_statistical_outlier_removal_deviation", 0.2,  0,  1)

gen.add("ransac_plane_segmentation_deviation",    double_t,    0, "ransac_plane_segmentation_deviation", 0.008,  0,  100)
gen.add("ransac_plane_segmentation_iterations",    double_t,    0, "ransac_plane_segmentation_deviation_iterations", 100,  0,  1000)

gen.add("euclidean_clustering_tolerance",    double_t,    0, "euclidean_clustering_tolerance",  0.015,  0,  3)
gen.add("euclidean_clustering_max",    double_t,    0, "euclidean_clustering_max", 200,  1, 1000)
gen.add("euclidean_clustering_min",    double_t,    0, "euclidean_clustering_min", 950,  1, 10000)



exit(gen.generate(PACKAGE, "object_detection", "pcl_parameters"))
